MvcApp/MvcApp/Areas/Users/Controllers/PostsController.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using MvcApp.Data;
+using MvcApp.Models;
+
+namespace MvcApp.Controllers
+{
+    [Area("Users")]
+    [Authorize]
+    public class PostsController : Controller
+    {
+        private readonly MvcAppContext _context;
+
+        public PostsController(MvcAppContext context)
+        {
+            _context = context;
+        }
+
+        // GET: Posts
+        public async Task<IActionResult> Index()
+        {
+            var curUserId=HttpContext.User.FindFirstValue(ClaimTypes.NameIdentifier);
+            return View(_context.Posts.Where(p=>p.AuthorId==curUserId));
+        }
+
+        // GET: Posts/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null || _context.Posts == null)
+            {
+                return NotFound();
+            }
+
+            var post = await _context.Posts
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (post == null)
+            {
+                return NotFound();
+            }
+
+            return View(post);
+        }
+
+        // GET: Posts/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: Posts/Create
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,Title,Content")] Post post)
+        {
+            Post curPost = new Post();
+            if (ModelState.IsValid)
+            {
```

Comment:
Post modelga yozgan commentlarimni resolve qiganizdan keyin, mana shu joyda sizda ModelState.IsValid false kelishni boshlidi. qanaqa qilib to'g'irlash mumkinligini Dilshod yoki Shoyusufdan so'rasez bo'ladi